subprojects/core/src/integTest/groovy/org/gradle/caching/internal/FinalizeBuildCacheConfigurationBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.test.fixtures.server.http.HttpBuildCache
+import org.junit.Rule
+import spock.lang.Unroll
+
+class FinalizeBuildCacheConfigurationBuildOperationIntegrationTest extends AbstractIntegrationSpec {
+
+    private static final SOME_CREDENTIALS =
+        """
+            remote.credentials {
+                username = "user"
+                password = "pass"
+            }
+        """
+
+    private static final NO_CREDENTIALS = ''
+
+    private static final INCOMPLETE_CREDENTIALS =
+        """
+            remote.credentials {
+                username = "user"
+            }
+        """
+
+    @Rule
+    BuildOperationsFixture buildOperations = new BuildOperationsFixture(executer, temporaryFolder)
+
+    @Rule
+    HttpBuildCache httpBuildCache = new HttpBuildCache(testDirectoryProvider)
+
+    def "local build cache configuration is exposed"() {
+        given:
+        def cacheDir = temporaryFolder.file("cache-dir").createDir()
+        settingsFile << """
+            buildCache {
+                local(DirectoryBuildCache) {
+                    enabled = true 
+                    directory = '${cacheDir.absoluteFile.toURI().toString()}'
+                    push = true 
+                }
+            }
+        """
+        executer.withBuildCacheEnabled()
+
+        when:
+        succeeds("help")
+
+        then:
+        def result = result()
+        result.local.className == 'org.gradle.caching.local.DirectoryBuildCache'
+        result.local.config.location == cacheDir.absoluteFile.toString()
+        result.local.type == 'directory'
+        result.local.push == true
+
+        result.remote == null
+    }
+
+    @Unroll
+    def "remote build cache configuration is exposed"() {
```

Comment:
ah, if this is the test that requires the http build cache in core, I think it's a better idea to move this part into build-cache-http